src/common/exponent/exponentHelper.ts

```diff
@@ -105,305 +74,199 @@ export class ExponentHelper {
             });
     }
 
-    public getExponentPackagerOptions(): Q.Promise<any> {
+    public getExpPackagerOptions(): Q.Promise<ExpConfigPackager> {
         this.lazilyInitialize();
-        return this.readFromExpJson<string>("packagerOpts");
+        return this.getFromExpConfig<ExpConfigPackager>("packagerOpts")
+            .then(opts => opts || {});
     }
 
-    /**
-     * File used as an entrypoint for exponent. This file's component should be registered as "main"
-     * in the AppRegistry and should only render a entrypoint component.
-     */
-    private createIndex(): Q.Promise<void> {
+    private createExpoEntry(name: string): Q.Promise<void> {
         this.lazilyInitialize();
-        const pkg = require("../../../package.json");
-        const extensionVersionNumber = pkg.version;
-        const extensionName = pkg.name;
-
-        return Q.all<string>([this.entrypointComponent(), this.entrypoint()])
-            .spread((componentName: string, entryPointFile: string) => {
-                const fileContents =
-                    `// This file is automatically generated by ${extensionName}@${extensionVersionNumber}
-// Please do not modify it manually. All changes will be lost.
-var React = require('${this.projectRootPath}/node_modules/react');
-var {Component} = React;
+        return this.detectEntry()
+            .then((entryPoint: string) => {
+                const content = this.generateFileContent(name, entryPoint);
+                return this.fs.writeFile(this.pathToFileInWorkspace(EXPONENT_INDEX), content);
+            });
 
-var ReactNative = require('${this.projectRootPath}/node_modules/react-native');
-var {AppRegistry} = ReactNative;
+    }
 
-var entryPoint = require('${this.projectRootPath}/${entryPointFile}');
+    private detectEntry(): Q.Promise<string> {
+        this.lazilyInitialize();
+        return Q.all([
+            this.fs.exists(this.pathToFileInWorkspace(DEFAULT_EXPONENT_INDEX)),
```

Comment:
Instead of checking some predefined "magic" locations you might want to look at `"main"` entry in `package.json`. I ddn't find any documentation but from my experience it always points to the main entry point. Also note the case when for example `"main"` points to non-existent file - in that case you'll need to check for ios and adroid variants of that file, i.e.

    index.js (doesn't exist) -> index.ios.js / index.android.js